### PR TITLE
[ISSUE #1502]♻️Refactor PullMessageRequestHeader🍻

### LINKED ISSUE #1502: [Refactor♻️]Refactor PullMessageRequestHeader
### Which part of the code will be refactored?

_No response_

### What problems does the refactor aim to solve?

_No response_

### Do you have any specific design or implementation considerations?

_No response_

### Is there a related issue or PR?

_No response_

### Checklist

- [ ] Ensure the refactor does not introduce new bugs

- [ ] Update unit tests if applicable

- [ ] Ensure the refactor does not negatively impact performance

- [ ] Document any new patterns or architecture changes

- [ ] Test the refactor in different environments (e.g., staging, production)

### PR DESCRIPTION
**Pull request by mxsm**
### Which Issue(s) This PR Fixes(Closes)

Fixes #1502 

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for required fields in message request processing.
	- Updated `queue_id` to be a mandatory field, improving data integrity.

- **Bug Fixes**
	- Removed potential silent failures by enforcing stricter checks on required fields.
	- Improved handling of `queue_id` to avoid potential panics.

- **Tests**
	- Added new test cases for serialization and deserialization processes, including edge cases for missing fields.

### CODE REVIEW

File: rocketmq-remoting/src/protocol/header/pull_message_request_header.rs [hunk 1]
```diff
@@ -484,10 +542,286 @@ impl TopicRequestHeaderTrait for PullMessageRequestHeader {
...
 
     fn queue_id(&self) -> Option<i32> {
-        self.queue_id
+        Some(self.queue_id)
     }
 
     fn set_queue_id(&mut self, queue_id: Option<i32>) {
-        self.queue_id = queue_id;
+        self.queue_id = queue_id.unwrap();
+    }
```
[the remark this answers is not in the document]
Comment (mxsm) [reply rocketmq-remoting/src/protocol/header/pull_message_request_header.rs:550]: I will do it later

**Review by TeslaRustor — APPROVED** [review]
